Extension/.scripts/common.ts

```diff
@@ -0,0 +1,290 @@
+import { chdir, cwd, env } from 'process';
+import { setImmediate } from 'timers/promises';
+import { promisify } from 'util';
+import { filepath } from '../src/Utility/Filesystem/filepath';
+import { is } from '../src/Utility/System/guards';
+import { verbose } from '../src/Utility/Text/streams';
+
+export const $root = resolve(`${__dirname}/..`);
+export let $cmd = 'main';
+export let $scenario = '';
+
+// loop thru the args and pick out --scenario=... and remove it from the $args and set $scenario
+export const $args  = process.argv.slice(2).filter(each => !(each.startsWith('--scenario=') && ($scenario = each.substring('--scenario='.length))));
+
+/** enqueue the call to the callback function to happen on the next available tick, and return a promise to the result */
+export function then<T>(callback: () => Promise<T>|T): Promise<T> {
+    return setImmediate().then(callback);
+}
+
+export const pwd = env.INIT_CWD ?? cwd();
+verbose(yellow(`pwd: ${pwd}`));
+
+// ensure we're in the extension folder.
+chdir($root);
+
+// dump unhandled async errors to the console and exit.
+process.on('unhandledRejection', (reason: any, _promise) => {
+    error(`${reason?.stack?.split(/\r?\n/).filter(l => !l.includes('node:internal') && !l.includes('node_modules')).join('\n')}`);
+    process.exit(1);
+});
+
+const git = new Command('git');
+export const Git = async (...args: Parameters<Awaited<CommandFunction>>) => (await git)(...args);
+export const GitClean = async (...args: Parameters<Awaited<CommandFunction>>) => (await new Command(await git, 'clean'))(...args);
+
+export async function getModifiedIgnoredFiles() {
+    const {code, error, stdio } = await GitClean('-Xd', '-n');
+    if (code) {
+        throw new Error(`\n${error.all().join('\n')}`);
+    }
+
+    // return the full path of files that would be removed.
+    return Promise.all(stdio.filter("Would remove").map((s) => filepath.exists(s.replace(/^Would remove /, ''), $root)).filter(p => p));
+}
+
+export async function rimraf(...paths: string[]) {
+    const all = [];
+    for (const each of paths) {
+        if (!each) {
+            continue;
+        }
+        if (await filepath.isFolder(each)) {
+            verbose(`Removing folder ${red(each)}`);
+            all.push(rm(each, {recursive: true, force: true}));
+            continue;
+        }
+        verbose(`Removing file ${red(each)}`);
+        all.push(rm(each, {force: true}));
+    }
+    await Promise.all(all);
+}
+
+export async function mkdir(filePath: string) {
+    const [fullPath, info] = await filepath.stats(filePath, $root);
+    if (info) {
+        if (info.isDirectory()) {
+            return fullPath;
+        }
+        throw new Error(`Cannot create directory '${filePath}' because thre is a file there.`);
+    }
+
+    await md(fullPath, { recursive: true });
+    return fullPath;
+}
+
+export const glob: (pattern: string, options?: IOptions | undefined) => Promise<string[]> = promisify(globSync);
```

Comment:
The formatting/spacing around `|` appears inconsistent is file. `IOptions | undefined` or `IOptions | undefined` or `string| Promise<string>`. Can that be added to the formatting spec? I'm not sure what the recommended spacing is.

Reply:
adding `"@typescript-eslint/space-infix-ops": "error",` rule into eslint.